app/shared/app-data/src/androidMain/kotlin/domain/torrent/service/AndroidTorrentServiceConnection.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2024-2025 OpenAni and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license, which can be found at the following link.
+ *
+ * https://github.com/open-ani/ani/blob/main/LICENSE
+ */
+
+package me.him188.ani.app.domain.torrent.service
+
+import android.content.BroadcastReceiver
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.ServiceConnection
+import android.os.IBinder
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.suspendCancellableCoroutine
+import me.him188.ani.app.domain.torrent.IRemoteAniTorrentEngine
+import me.him188.ani.app.domain.torrent.LifecycleAwareTorrentServiceConnection
+import me.him188.ani.app.domain.torrent.TorrentServiceConnection
+import me.him188.ani.utils.logging.debug
+import me.him188.ani.utils.logging.error
+import me.him188.ani.utils.logging.warn
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resume
+import kotlin.time.Duration.Companion.minutes
+
+/**
+ * 管理与 [AniTorrentService] 的连接并获取 [IRemoteAniTorrentEngine] 远程访问接口.
+ * 通过 [getBinder] 获取服务接口, 再启动完成并绑定之前将挂起协程.
+ *
+ * 服务连接控制依赖的 lifecycle 应当尽可能大, 所以应该使用
+ * [ProcessLifecycleOwner][androidx.lifecycle.ProcessLifecycleOwner]
+ * 或其他可以涵盖 app 全局生命周期的自定义 [LifecycleOwner] 来管理服务连接.
+ * 不能使用 [Activity][android.app.Activity] (例如 [ComponentActivity][androidx.core.app.ComponentActivity])
+ * 的生命周期, 因为在屏幕旋转 (例如竖屏转全屏播放) 的时候 Activity 可能会摧毁并重新创建,
+ * 这会导致 [AndroidTorrentServiceConnection] 错误地重新绑定服务或重启服务.
+ *
+ * @see androidx.lifecycle.ProcessLifecycleOwner
+ * @see ServiceConnection
+ * @see AniTorrentService.onStartCommand
+ * @see me.him188.ani.android.AniApplication
+ */
+class AndroidTorrentServiceConnection(
+    private val context: Context,
+    private val onRequiredRestartService: () -> ComponentName?,
+    parentCoroutineContext: CoroutineContext = Dispatchers.Default,
+) : ServiceConnection,
+    LifecycleAwareTorrentServiceConnection<IRemoteAniTorrentEngine>(parentCoroutineContext) {
+    private val startupIntentFilter by lazy { IntentFilter(AniTorrentService.INTENT_STARTUP) }
+    private val acquireWakeLockIntent by lazy {
+        Intent(context, AniTorrentService.actualServiceClass).apply {
+            putExtra("acquireWakeLock", 1.minutes.inWholeMilliseconds)
+        }
+    }
```

Comment:
Is lazy really needed here?

Lazy introduces synchornization overhead (which is on main thread).